app_spec/zomes/blog/code/src/blog.rs

```diff
@@ -45,100 +45,100 @@ pub fn handle_check_sum(num1: u32, num2: u32) -> JsonString {
 }
 
 pub fn handle_hash_post(content: String) -> JsonString {
-    Entry::App(
+    let post_entry = Entry::App(
         AppEntryType::from("post"),
         AppEntryValue::from(
             Post::new(
                 &content,
                 "now",
             ),
         )
-    ).into()
+    );
 
-    // match hdk::entry_address(&post_entry) {
-    //     Ok(address) => AddressResponse{address}.into(),
-    //     Err(hdk_error) => hdk_error.into(),
-    // }
+    match hdk::entry_address(&post_entry) {
+        Ok(address) => AddressResponse{address}.into(),
+        Err(hdk_error) => hdk_error.into(),
+    }
```

Comment:
How about some declarative programming here:
```rust
hdk::entry_address(&post_entry)
    .map(|address| AddressResponse { address }.into())
    .unwrap_or_else(|hdk_err| hdk_err.into())
```

Reply:
pattern matching doesn't fall under declarative programming?

Reply:
http://wiki.c2.com/?PatternMatching

> A feature of FunctionalProgramming and LogicProgramming languages ... which is fundamental to making these programming paradigms declarative

Reply:
Okay, it's more using the tools, the standard library supplies instead of reinventing the wheel over and over again. It makes it easier to reason about the code

Reply:
In that particular case the code looks almost identical and just as easy to reason about!